### PR TITLE
LPS-199537 Support picklist object fields in unique composite key

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://liferay.atlassian.net/browse/LPS-199537

### CODE REVIEW

**Comment by mateussgn** [comment]
Blocked by: https://github.com/liferay-objects/liferay-portal/pull/2688

**Comment by mateussgn** [comment]
Blocked by: https://github.com/liferay-objects/liferay-portal/pull/2688

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Review by pedro-oliveira446 — APPROVED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java [hunk 1]
```diff
@@ -125,11 +127,24 @@ private Predicate _getPredicate(
...
+			else {
+				value = String.valueOf(entryValues.get(objectField.getName()));
+			}
+
 			Predicate uniqueCompositeKeyObjectFieldPredicate =
 				ObjectEntrySearchUtil.getUniqueCompositeKeyObjectFieldPredicate(
-					(Column<?, Object>)column, objectField.getDBType(),
-					GetterUtil.getString(
-						entryValues.get(objectField.getName())));
+					(Column<?, Object>)column, objectField.getDBType(), value);
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java:147]: @mateussgn is this the only new code in this PR? If yes, it is approved 😄

Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java:147]: Yes, it is! Thank you, @carolmariaabb!

**Comment by mateussgn** [comment]
ci:reevaluate:1267401_11609

**Comment by mateussgn** [comment]
ci:reevaluate:1257401_3889

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:forward:force

**Comment by mateussgn** [comment]
ci:forward:force

**Pull request closed without merging**